server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -109,6 +106,7 @@
 {
   private static final EmittingLogger log = new EmittingLogger(RealtimePlumber.class);
   private static final int WARN_DELAY = 1000;
+  private static final int MAX_ROW_EXCEED_CHECK_COUNT = 100000;
```

Comment:
Changed to `Math.min(config.getMaxRowsInMemory() >> 2, MAX_ROW_EXCEED_CHECK_COUNT)`. Thanks.